app-modules/engagement/src/Enums/EngagementDeliveryMethod.php

```diff
@@ -46,8 +48,39 @@ enum EngagementDeliveryMethod: string implements HasLabel
     public function getLabel(): ?string
     {
         return match ($this) {
-            static::Email => 'Email',
-            static::Sms => 'SMS',
+            EngagementDeliveryMethod::Email => 'Email',
+            EngagementDeliveryMethod::Sms => 'SMS',
         };
     }
+
+    public function getDefault(): EngagementDeliveryMethod
+    {
+        return EngagementDeliveryMethod::Email;
+    }
+
+    public function getCaseDisabled(): bool
+    {
+        return match ($this) {
+            EngagementDeliveryMethod::Sms => Integration::Twilio->isOff(),
+            default => false,
+        };
+    }
+
+    public function getLabelForIntegrationState(): string
+    {
+        return $this->getCaseDisabled()
+            ? sprintf(
+                '%s (%s)',
+                $this->getLabel(),
+                IntegrationException::make(Integration::Twilio)->getMessage()
```

Comment:
Should we have a match statement here to future proof this? If we add another integration to `getCaseDisabled` in the future it would continue to display the Twilio error wouldn't it?